test/connectors/httpParsers/PenaltiesHttpParserSpec.scala

```diff
@@ -0,0 +1,135 @@
+import models.penalties.PenaltiesSummary
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+import play.api.libs.json.{JsObject, Json}
+import play.api.http.Status
+import models.errors._
+import uk.gov.hmrc.http.HttpResponse
+
+class PenaltiesHttpParserSpec extends AnyWordSpecLike with Matchers {
+
+  "PenaltiesHttpParser" when {
+
+    "the http response status is 200 OK" should {
+      val jsonResponse = Json.obj(
+        "noOfPoints" -> 3,
+        "noOfEstimatedPenalties" -> 2,
+        "noOfCrystalisedPenalties" -> 1,
+        "estimatedPenaltyAmount" -> 123.45,
+        "crystalisedPenaltyAmountDue" -> 54.32,
+        "hasAnyPenaltyData" -> true
+      )
+      val httpResponse = HttpResponse(Status.OK, jsonResponse.toString())
+      val expected = Right(PenaltiesSummary(
+        noOfPoints = 3,
+        noOfEstimatedPenalties = 2,
+        noOfCrystalisedPenalties = 1,
+        estimatedPenaltyAmount = 123.45,
+        crystalisedPenaltyAmountDue = 54.32,
+        hasAnyPenaltyData = true
+      ))
+      val result = PenaltiesReads.read("", "", httpResponse)
+
+      "return a PenaltiesSummaryInstance" in {
+        result shouldBe expected
+      }
+    }
+
+    "the http response status is 404 NOT_FOUND" should {
+      val httpResponse = HttpResponse(Status.NOT_FOUND, "")
+      val expected = Right(PenaltiesSummary.empty)
+      val result = PenaltiesReads.read("", "", httpResponse)
+
+      "return a 404 error" in {
+        result shouldEqual expected
+      }
+    }
+
+    "the http response status is 400 BAD_REQUEST (single error)" should {
+      val httpResponse = HttpResponse(Status.BAD_REQUEST,
+        Json.obj(
+          "code" -> "VRN_INVALID",
+          "reason" -> "Fail"
+        ).toString()
+      )
+      val expected = Left(BadRequestError(
+        code = "VRN_INVALID",
+        errorResponse = "Fail"
+      ))
+      val result = PenaltiesReads.read("", "", httpResponse)
+
+      "return a BadRequestError" in {
+        result shouldEqual expected
+      }
+    }
+
+    "the http response status is 400 BAD_REQUEST (unknown API error json)" should {
+
+      val httpResponse = HttpResponse(Status.BAD_REQUEST,
+        Json.obj(
+          "foo" -> "VRN_INVALID",
+          "bar" -> "Fail"
+        ).toString()
+      )
+      val expected = Left(UnknownError)
+
+      val result = PenaltiesReads.read("", "", httpResponse)
+
+      "return a UnknownError" in {
+        result shouldEqual expected
+      }
+    }
+
+    "the HTTP response status is 5xx" should {
+      val body: JsObject = Json.obj(
+        "code" -> "GATEWAY_TIMEOUT",
+        "message" -> "GATEWAY_TIMEOUT"
+      )
+
+      val httpResponse = HttpResponse(Status.GATEWAY_TIMEOUT, body.toString())
+      val expected = Left(ServerSideError(Status.GATEWAY_TIMEOUT.toString, httpResponse.body))
+      val result = DirectDebitStatusReads.read("", "", httpResponse)
```

Comment:
Should this not be PenaltiesReads?

Reply:
Yes it should, making the change and will push up